workflows/handle_data.nf

```diff
@@ -34,27 +34,54 @@ workflow HANDLE_DATA {
     json = Channel.fromPath( input_json )
 
     // read the json and create many json as there are combinations of noisers and splitters. the message_from_check is passed only to enforce that this modules does not run untill check_module is finished.
-    INTERPRET_JSON(json, message_from_check)
+    INTERPRET_JSON( json, message_from_check )
 
-    // for each user json many json wiil be created in the same process, but then we need to parallelize and work on them one by one.
-    parsed_json = INTERPRET_JSON.out.interpreted_json.flatten()
-    
-    // launch splitting subworkflow 
-    SPLIT_CSV(csv, parsed_json)
+    // the above process outputs three channels one with all the information (split+ transform), one with only split info, and one with only transform. Each of this channels have to be transformed into a tuple with a common unique id for a given combination.
+    experiment_json = INTERPRET_JSON.out.experiment_json.flatten().map{
+        it -> ["${it.baseName}".split('-')[0..-2].join("-"), it]
+    }
 
-    // launch the actual noise subworkflow
-    TRANSFORM_CSV( SPLIT_CSV.out.split_data )
+    // the split has only the keyword to match to the transform
+    split_json = INTERPRET_JSON.out.split_json.flatten().map{
+        it -> ["${it.baseName}".split('-')[-1], it]
+    }
+
+    // and transform has both keys to match to everything toghether
+    transform_json = INTERPRET_JSON.out.transform_json.flatten().map{
+        it -> ["${it.baseName}".split('-')[-1], "${it.baseName}".split('-')[0..-3].join("-"), it]
+    }
+
+    // run split with json that only contains experiment name and split information. It runs only the necessary times, all unique ways to split + default split (random split) or column split (already present in data).
+    SPLIT_CSV( csv, split_json )
+
+    // assign to each splitted data the associated ransform information based on the split_transform_key generated in the interpret step. 
+    transform_split_tuple = transform_json.combine( SPLIT_CSV.out.split_data, by: 0 )
 
-    // Launch the shuffle, (always happening on default) and disjointed from split and noise. Data are randomly splitted into this module already.
-    // it takes a random json from those interpreted so that is dependant on that process and to have the experiment name key, used later in the train step.
-    SHUFFLE_CSV(csv, parsed_json.first())
+    // launch the actual noise subworkflow
+    TRANSFORM_CSV( transform_split_tuple )
 
-    // merge output of shuffle to the output of noise
-    data = TRANSFORM_CSV.out.transformed_data.concat(SHUFFLE_CSV.out.shuffle_data)
+    // unify transform output with interpret experiment json. so that each final data has his own fingerprint json that generated it + keyword. drop all other non relevant fields.
+    tmp = experiment_json.combine( TRANSFORM_CSV.out.transformed_data, by: 0 ).map{
+        it -> ["${it[6].name} - ${it[0]}", it[2], it[1], it[4]]
```

Comment:
could you add a comment of what is it[2], it[1], it[4] 
Otherwise i don't really understand what they are without following thoroughtfully the pipeline code